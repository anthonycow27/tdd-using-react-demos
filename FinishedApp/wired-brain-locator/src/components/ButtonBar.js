import React, { Component } from 'react';
import LocationButton from '../components/LocationButton.js';
import './ButtonBar.css';

class ButtonBar extends Component {



    render(){
        var shopArray = this.props.shops;
        const buttons = shopArray.map(function(shop,id){
            return (
                <LocationButton
                    shop = {shop}
                    key = {id}
                />);
        });
        return(<div>{buttons}</div>);
    }
}

export default ButtonBar;