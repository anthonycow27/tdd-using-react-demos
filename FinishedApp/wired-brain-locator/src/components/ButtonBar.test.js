import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ButtonBar from './ButtonBar';



describe('ButtonBar', () => {
        const shops=["location1","location2"];

    it('renders without crashing', () => {
        const component = shallow(<ButtonBar shops={shops}/>);
        expect(component).toBeDefined();
    });

    it('matches the snapshot', () => {
        const component = shallow(<ButtonBar shops={shops}/>);
        expect(toJson(component)).toMatchSnapshot();
    });

});