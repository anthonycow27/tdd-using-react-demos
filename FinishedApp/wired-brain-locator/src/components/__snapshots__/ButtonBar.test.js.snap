// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonBar matches the snapshot 1`] = `
<div>
  <LocationButton
    key="0"
    shop="location1"
  />
  <LocationButton
    key="1"
    shop="location2"
  />
</div>
`;
